lib/rules/id-blacklist.js

```diff
@@ -34,7 +34,7 @@ module.exports = {
         // Helpers
         //--------------------------------------------------------------------------
 
-        var blacklist = context.options;
+        var blacklistRegex = new RegExp("^(" + context.options.join("|") + ")$");
```

Comment:
You have to differentiate no matter what, considering that `$` is both a valid variable name and a regex special character.

One other we could do this without worrying about regex vs. string ambiguity is require `/` surrounding cases that should be considered regexes, which is safe considering valid identifiers can't contain `/`.

`"id-blacklist": [2, "foobar", "/foo$/", "foo$"]` would match `foobar`, `foo$`, and any id ending in `foo`.

And no matter what we choose to signify, need to regex escape the string literals if they end up inside a regex constructor.